src/python/m5/util/convert.py

```diff
@@ -258,6 +264,16 @@ def toMemoryBandwidth(value):
 
 
 def toMemorySize(value):
+    if (
+        type(value) is str
+        and len(value) > 1
+        and value[-2] in binary_prefixes.keys()
+        and not "i" in value
+    ):
+        print(
+            f"warn: Base 10 memory/cache size {value} will be cast to base 2"
+            + f" size {value[0:-2]}{value[-2].upper()}i{value[-1]}"
+        )
```

Comment:
Here's a cheap solution to your circular importing. 

You can import anywhere so just do it when execute this function, not when the module is loaded.

```suggestion
        from m5.util import warn
        warn(
            f"Base 10 memory/cache size {value} will be cast to base 2"
            + f" size {value[0:-2]}{value[-2].upper()}i{value[-1]}"
        )
```